src/main/java/io/gravitee/policy/groovy/model/message/ScriptableMessage.java

```diff
@@ -0,0 +1,195 @@
+        return correlationId();
+    }
+
+    @Override
+    public String parentCorrelationId() {
+        return message.parentCorrelationId();
+    }
+
+    public String getParentCorrelationId() {
+        return parentCorrelationId();
+    }
+
+    @Override
+    public long timestamp() {
+        return message.timestamp();
+    }
+
+    public long getTimestamp() {
+        return timestamp();
+    }
+
+    @Override
+    public boolean error() {
+        return message.error();
+    }
+
+    public boolean isError() {
+        return error();
+    }
+
+    @Override
+    public Message error(boolean error) {
+        return message.error(error);
+    }
+
+    @Override
+    public Map<String, Object> metadata() {
+        return message.metadata();
+    }
+
+    public Map<String, Object> getMetadata() {
+        return metadata();
+    }
+
+    @Override
+    public HttpHeaders headers() {
+        return message.headers();
+    }
+
+    public ScriptableHttpHeaders getHeaders() {
+        return new ScriptableHttpHeaders(headers());
+    }
+
+    @Override
+    public Message headers(HttpHeaders headers) {
+        return message.headers(headers);
+    }
+
+    @Override
+    public Buffer content() {
+        return message.content();
+    }
+
+    public String getContent() {
+        return message.content() == null ? "" : content().toString();
+    }
+
+    @Override
+    public Message content(Buffer content) {
+        return message.content(content);
+    }
+
+    @Override
+    public Message content(String content) {
+        return message.content(content);
+    }
+
+    @Override
+    public void ack() {
+        message.ack();
+    }
+
+    @Override
+    public <T> T attribute(String name) {
+        return message.attribute(name);
+    }
+
+    public <T> T getAttribute(String name) {
+        return attribute(name);
+    }
+
+    @Override
+    public <T> List<T> attributeAsList(String name) {
+        return message.attributeAsList(name);
+    }
+
+    public <T> List<T> getAttributeAsList(String name) {
+        return attributeAsList(name);
+    }
+
+    @Override
+    public Message attribute(String name, Object value) {
+        return message.attribute(name, value);
+    }
+
+    @Override
+    public Message removeAttribute(String name) {
+        return message.removeAttribute(name);
+    }
+
+    @Override
+    public Set<String> attributeNames() {
+        return message.attributeNames();
+    }
+
+    @Override
+    public <T> Map<String, T> attributes() {
+        return message.attributes();
+    }
+
+    public Map<String, Object> getAttributes() {
+        return new ScriptableMessageAttributes(message);
+    }
+
+    @Override
+    public <T> T internalAttribute(String name) {
+        return message.internalAttribute(name);
+    }
+
+    @Override
+    public Message internalAttribute(String name, Object value) {
+        return message.internalAttribute(name, value);
```

Comment:
Internal attributes should not be accessible from Groovy imho.

Reply:
✅